### PR TITLE
Applayer plugin 5053 v3.13

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/5053

Describe changes:
- get ready to use dynamic number of app-layer protos (also work with static constant) in some places
- preventive fix of macro with parenthesis cc @jufajardini 

#12116 with clean history and in-tree plugin testing (without logging enabled)

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.tls.parser | 1152 | 1203 | 104.43% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.tx.ftp | 95372 | 102159 | 107.12% |
| .ftp.memuse | 2878 | 10638 | 369.63% | 

Pipeline 23315

File: examples/plugins/zabbix/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "suricata-zabbix"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
```
Comment (jasonish) [inline examples/plugins/zabbix/Cargo.toml:7]: I think this is correct for using the C ABI?  @chifflier?

Comment (victorjulien) [reply examples/plugins/zabbix/Cargo.toml:7]: @chifflier this may interest you https://github.com/OISF/suricata/pull/12062

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/12127

**Pull request closed without merging**